src/main/java/com/github/emcat/metric_calculators/MethodMetricsAggregator.java

```diff
@@ -0,0 +1,75 @@
+package com.github.emcat.metric_calculators;
+
+import net.sourceforge.pmd.lang.Parser;
+import net.sourceforge.pmd.lang.ParserOptions;
+import net.sourceforge.pmd.lang.java.JavaLanguageModule;
+import net.sourceforge.pmd.lang.java.ast.*;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+
+public class MethodMetricsAggregator {
+    public MethodMetricsData calculateSingleMethodMetrics(MethodDescriptor methodDescriptor) {
+        try {
+            exception = null;
+
+            ASTMethodDeclaration methodDeclaration = getMethodDeclaration(
+                    methodDescriptor.filepath(),
+                    methodDescriptor.className(),
+                    methodDescriptor.methodName()
+            );
+
+            return metricsCalculator.calculateMethodMetrics(methodDeclaration);
+        }
+        catch (Exception exception) {
+            this.exception = new MethodProcessingException(methodDescriptor, exception);
+            return null;
+        }
+    }
+
+    public MethodProcessingException getException() {
+        return exception;
+    }
+
+    private ASTMethodDeclaration getMethodDeclaration(String filename, String className, String methodName) throws IOException {
+        ASTClassOrInterfaceDeclaration classDeclaration = getClassDeclaration(filename, className);
+        for (ASTAnyTypeBodyDeclaration classInnerDeclaration : classDeclaration.getDeclarations()) {
+            if (classInnerDeclaration.getKind() == ASTAnyTypeBodyDeclaration.DeclarationKind.METHOD) {
+                ASTMethodDeclaration methodDeclaration = (ASTMethodDeclaration) classInnerDeclaration.getDeclarationNode();
+                if (methodDeclaration.getName().equals(methodName)) {
+                    return methodDeclaration;
+                }
+            }
+        }
+
+        throw new RuntimeException(String.format("There is no method named %s in class named %s in file %s", methodName, className, filename));
```

Comment:
Same about exceptions as below, bad idea to build your logic using it. But if you really want to build your logic using that - at least create your own runtime exception, or choose suitable from existing